test/WebSites/ControllersFromServicesWebSite/Startup.cs

```diff
@@ -25,10 +25,15 @@ public void ConfigureServices(IServiceCollection services)
                 .AddMvc()
                 .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Clear())
                 .AddApplicationPart(typeof(TimeScheduleController).GetTypeInfo().Assembly)
-                .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Add(new TypesPart(
-                    typeof(AnotherController),
-                    typeof(ComponentFromServicesViewComponent),
-                    typeof(InServicesTagHelper))))
+                .ConfigureApplicationPartManager(manager =>
+                {
+                    manager.ApplicationParts.Add(new TypesPart(
+                      typeof(AnotherController),
+                      typeof(ComponentFromServicesViewComponent),
+                      typeof(InServicesTagHelper)));
+
+                    manager.FeatureProviders.Add(new AssemblyMetadataReferenceFeatureProvider());
```

Comment:
What is the minimal code a user would write for the #4497 scenario? User wants an assembly (containing tag helpers) available when compiling a view. I think this PR provides a solution for that requirement -- without an additional compilation callback or exposing the `MetadataReference` collection directly. But I'm hoping our users won't need to write a custom `IApplicationFeatureProvider<MetadataReferenceFeature>` implementation for this.

BTW feel free to poach #4497 if this PR does provide its fix.